pkg/roachprod/promhelperclient/client_test.go

```diff
@@ -57,8 +58,19 @@ func TestUpdatePrometheusTargets(t *testing.T) {
 			require.Equal(t, getUrl(promUrl, "c1"), url)
 			ir, err := getInstanceConfigRequest(io.NopCloser(body))
 			require.Nil(t, err)
-			// TODO (bhaskar): check for the correct yaml
 			require.NotNil(t, ir.Config)
+			configs := make([]*CCParams, 0)
+			require.Nil(t, yaml.UnmarshalStrict([]byte(ir.Config), &configs))
+			require.Equal(t, 2, len(configs))
+			for _, c := range configs {
+				if c.Labels["node"] == "1" {
```

Comment:
nit: something like this might be simpler/easier to expand if you ever add more nodes to this test case?

```
nodes := map[int]string{1: "n1", 3: "n3"}
c.httpPut = func(ctx context.Context, url string, h *http.Header, body io.Reader) (
...
      for _, c := range configs {
				nodeID, err := strconv.Atoi(c.Labels["node"])
				require.NoError(t, err)
				require.Equal(t, nodes[nodeID], c.Targets[0])
				c.Labels["tenant"] = "system"
			}
...
err := c.UpdatePrometheusTargets(ctx, promUrl, "c1", false, nodes, true, l)
```